addon/bootstrap.js

```diff
@@ -43,11 +49,16 @@ this.Bootstrap = {
     this.variation = variation;
     this.reason = reason;
 
-    // if addon was just installed, check if user is eligible
-    if ((this.REASONS[reason]) === "ADDON_INSTALL") {
+    // Check if the user is eligible to run this study using the |isEligible|
+    // function when the study is initialized (install or upgrade, the latter
+    // being interpreted as a new install).
+    if (
+      reason === this.REASONS.ADDON_INSTALL ||
+      reason === this.REASONS.ADDON_UPGRADE
```

Comment:
This is the only change I'm not certain about, but since here it is only affecting the `isEligible` check, I'll merge it for now. I [commented](https://github.com/mozilla/shield-studies-addon-template/pull/49/commits/1e9ca3d9bd79d8fd482f2b42e572810b1a7879b3#r168928701) on this change for your PR in the template repo too. Would be good to confirm with you and Gregg on how resetting the study on upgrade affects the study's behavior, if at all, in deployment.